docs/source/custom_sockets.rst

```diff
@@ -0,0 +1,214 @@
+
+    # or add new:
+
+    custom_socket.add_endpoint(new_endpoint)  # see above code for endpoint examples;
+    custom_socket.add_dependency(new_dependency)  # see above code for dependency examples;
+
+    custom_socket.update()
+
+
+Running the custom socket
+-------------------------
+
+To run custom socket use::
+
+    # this will run the my_endpoint - and call the custom_script (method is GET);
+    result = custom_socket.run(method='GET', endpoint_name='my_endpoint')
+
+
+Read all endpoints
+------------------
+
+To get the all defined endpoints in custom socket run::
+
+    endpoints = custom_socket.get_endpoints()
+
+    for endpoint in endpoints:
+        print(endpoint.name)
+        print(endpoint.calls)
+
+To run particular endpoint::
+
+    endpoint.run(method='GET')
+    # or:
+    endpoint.run(method='POST', data={'name': 'test_name'})
+
+The data will be passed to the api call in the request body.
+
+Custom sockets endpoints
+------------------------
+
+Each custom socket is created from at least one endpoint. The endpoint is characterized by name and
+defined calls. Calls is characterized by name and methods. The name is a identification for dependency, eg.
+if it's equal to 'my_script' - the Script with label 'my_script' will be used (if exist and the source match),
+or new one will be created.
+There's a special wildcard method: `methods=['*']` - this mean that any request with
+any method will be executed in this endpoint.
+
+To add endpoint to the custom_socket use::
+
+    my_endpoint = Endpoint(name='my_endpoint')  # again - no api call here
+    my_endpoint.add_call(ScriptCall(name='custom_script'), methods=['GET'])
+    my_endpoint.add_call(ScriptCall(name='another_custom_script'), methods=['POST'])
+
+    custom_socket.add_endpoint(my_endpoint)
+
+Custom socket dependency
+------------------------
+
+Each custom socket has dependency - this is a meta information for endpoint: which resource
+should be used to return the api call results. The dependencies are bind to the endpoints call objects.
+Currently supported dependency in only script.
+
+**Using new script**
+
+::
+
+    custom_socket.add_dependency(
+        ScriptDependency(
+            Script(
+                label='custom_script',
+                runtime_name=RuntimeChoices.PYTHON_V5_0,
+                source='print("custom_script")'
+            )
+        )
+    )
+
+
+**Using defined script**
+
+::
+
+    another_custom_script = Script.please.get(id=2)
+    custom_socket.add_dependency(
+        ScriptDependency(
+            another_custom_script
+        )
+    )
+
+
+Custom socket recheck
+---------------------
+
+The creation of the socket can fail - this happen, eg. when endpoint name is already taken by another
+custom socket. To check the statuses use::
+
+    print(custom_socket.status)
+    print(custom_socket.status_info)
+
+There is a possibility to re-check socket - this mean that if conditions are met - the socket will be
```

Comment:
Also, dependencies can be mistakenly deleted. So re-check is useful here to well, "reinstall" everything that is missing. The socket itself is not actually created again as it's already there. It just checks endpoints and dependencies again.